src/scripts/clipperUI/ratingsHelper.ts

```diff
@@ -0,0 +1,538 @@
+										panel.setState({
+											userSelectedRatingsPromptStage: RatingsPromptStage.END
+										});
+									}
+								});
+							});
+						}
+					});
+				break;
+			case RatingsPromptStage.RATE:
+				let rateUrl: string = RatingsHelper.getRateUrlIfExists(clientType);
+				if (!Utils.isNullOrUndefined(rateUrl) && rateUrl.length > 0) {
+					buttons.push({
+						id: Constants.Ids.ratingsButtonRateYes,
+						label: Localization.getLocalizedString("WebClipper.Label.Ratings.Button.Rate"),
+						handler: () => {
+							window.open(rateUrl, "_blank");
+
+							panel.setState({
+								userSelectedRatingsPromptStage: RatingsPromptStage.END
+							});
+						}
+					}, {
+						id: Constants.Ids.ratingsButtonRateNo,
+						label: Localization.getLocalizedString("WebClipper.Label.Ratings.Button.NoThanks"),
+						handler: () => {
+							panel.setState({
+								userSelectedRatingsPromptStage: RatingsPromptStage.NONE
+							});
+						}
+					});
+				} else {
+					// this shouldn't happen
+					panel.setState({
+						userSelectedRatingsPromptStage: RatingsPromptStage.NONE
+					});
+				}
+				break;
+			case RatingsPromptStage.FEEDBACK:
+				let feedbackUrl: string = RatingsHelper.getFeedbackUrlIfExists(clipperState);
+				if (!Utils.isNullOrUndefined(feedbackUrl) && feedbackUrl.length > 0) {
+					buttons.push({
+						id: Constants.Ids.ratingsButtonFeedbackYes,
+						label: Localization.getLocalizedString("WebClipper.Label.Ratings.Button.Feedback"),
+						handler: () => {
+							window.open(feedbackUrl, "_blank");
+
+							panel.setState({
+								userSelectedRatingsPromptStage: RatingsPromptStage.END
+							});
+						}
+					}, {
+						id: Constants.Ids.ratingsButtonFeedbackNo,
+						label: Localization.getLocalizedString("WebClipper.Label.Ratings.Button.NoThanks"),
+						handler: () => {
+							panel.setState({
+								userSelectedRatingsPromptStage: RatingsPromptStage.NONE
+							});
+						}
+					});
+				} else {
+					// this shouldn't happen
+					panel.setState({
+						userSelectedRatingsPromptStage: RatingsPromptStage.NONE
+					});
+				}
+				break;
+			default:
+			case RatingsPromptStage.END:
+			case RatingsPromptStage.NONE:
+				break;
+		}
+
+		return buttons;
+	}
+
+	/**
+	 * We will show the ratings prompt if ALL of the below applies:
+	 *   * Ratings prompt is enabled for the ClientType/ClipperType
+	 *   * If StorageKeys.doNotPromptRatings is not "true"
+	 *   * If RatingsHelper.badRatingTimingDelayIsOver(...) returns true when provided StorageKeys.lastBadRatingDate
+	 *   * If RatingsHelper.badRatingVersionDelayIsOver(...) returns true when provided StorageKeys.lastBadRatingVersion and StorageKeys.lastSeenVersion
+	 *   * If RatingsHelper.clipSuccessDelayIsOver(...) returns true when provided StorageKeys.numClipSuccess
+	 */
+	public static shouldShowRatingsPrompt(clipperState: ClipperState): Promise<boolean> {
+		return new Promise<boolean>((loggingResolve, loggingReject) => {
```

Comment:
I'm not sure about this. I view public method comments like this as a description for the user/caller of this method (especially since when you hover over a method in VS Code etc., this is the description provided), and so the caller need not know about implementation details like that. #WontFix for now